cli/main.rs

```diff
@@ -102,17 +145,18 @@ fn main() {
   let should_display_info = flags.info;
 
   let state = ThreadSafeState::new(flags, rest_argv, ops::op_selector_std);
-  let mut main_worker = Worker::new(
+  let mut worker = Worker::new(
     "main".to_string(),
     startup_data::deno_isolate_init(),
     state.clone(),
   );
 
-  let main_future = lazy(move || {
-    // Setup runtime.
-    js_check(main_worker.execute("denoMain()"));
+  // TODO(ry) somehow combine the two branches below. They're very similar but
+  // it's difficult to get the types to workout.
```

Comment:
Maybe split those conditional branches into separate functions? Then combined with clap's parsing we'd get:
```rust
  match matches.subcommand() {
    ("eval", Some(info_match)) => {
      eval_command(...);
    }
    ("info", Some(info_match)) => {
       info_command(...);
    }
    ("fmt", Some(fmt_match)) => {
       fmt_command(...);
    }
    (script, Some(script_match)) => {
      run_script_command();
    }
    _ => {}
  }
```
I can do that in follow-up PR

Reply:
That looks like a great way to organize it.